service/worker/stream.go

```diff
@@ -153,40 +147,54 @@ func (ps *PushStream) processPushStreamResponse(ctx context.Context, subModel *s
 			workerMessagesNAckd.WithLabelValues(env, subModel.ExtractedTopicName, subModel.ExtractedSubscriptionName, subModel.PushEndpoint).Inc()
 		}
 
+		// discard response.Body after usage and ignore errors
+		_, err = io.Copy(ioutil.Discard, resp.Body)
+		err = resp.Body.Close()
+		if err != nil {
+			logger.Ctx(ps.ctx).Errorw("worker: push response error on response io close()", "status", resp.StatusCode, "subscription", ps.subcriptionName, "subscriberId", ps.subs.GetID(), "error", err.Error())
+		}
+
 		// TODO: read response body if required by publisher later
 	}
 }
 
-func (ps *PushStream) nack(_ context.Context, message *metrov1.ReceivedMessage) {
+func (ps *PushStream) nack(ctx context.Context, message *metrov1.ReceivedMessage) {
 	logger.Ctx(ps.ctx).Infow("worker: sending nack request to subscriber", "ackId", message.AckId, "subscription", ps.subcriptionName, "subscriberId", ps.subs.GetID())
 	ackReq := subscriber.ParseAckID(message.AckId)
 	// deadline is set to 0 for nack
 	modackReq := subscriber.NewModAckMessage(ackReq, 0)
-	ps.subs.GetModAckChannel() <- modackReq
+	// check for closed channel before sending request
+	if ctx.Err() == nil && ps.subs.GetModAckChannel() != nil {
```

Comment:
nil check not required now, since channel is being closed by same go routine

Reply:
done